src/main/java/yapper/Ui.java

```diff
@@ -0,0 +1,133 @@
+package yapper;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+/**
+ * Handles the printing of messages in response to the input form the user.
+ */
+public class Ui {
+    private static final String LINE = "    ____________________________________________________________\n";
+    private static final String INDENT = "    ";
+    private boolean hasEnded;
+    private final TaskList mainTasks;
+
+    /**
+     * Initialises a Ui that corresponds to a main {@link TaskList}.
+     * Main {@link TaskList} will also contain a reference to this instance.
+     *
+     * @param mainTasks Main {@link TaskList} to read from.
+     */
+    public Ui(TaskList mainTasks) {
+        this.mainTasks = mainTasks;
+        hasEnded = false;
+        mainTasks.setUi(this);
+    }
+
+    public static String indent() {
+        return INDENT;
+    }
+
+    public static String line() {
+        return LINE;
+    }
+
+    /**
+     * Greets the user with an introductory message.
+     */
+    public static void hello() {
+        String logo =
+                "       :::   :::           :::        :::::::::       :::::::::       ::::::::::       :::::::::\n"
+                + "      :+:   :+:         :+: :+:      :+:    :+:      :+:    :+:      :+:              :+:    :+:\n"
+                + "      +:+ +:+         +:+   +:+     +:+    +:+      +:+    +:+      +:+              +:+    +:+\n"
+                + "      +#++:         +#++:++#++:    +#++:++#+       +#++:++#+       +#++:++#         +#++:++#:\n"
+                + "      +#+          +#+     +#+    +#+             +#+             +#+              +#+    +#+\n"
+                + "     #+#          #+#     #+#    #+#             #+#             #+#              #+#    #+#\n"
+                + "    ###          ###     ###    ###             ###             ##########       ###    ###\n\n";
+        System.out.print(LINE + "    What's poppin' fam, it's ya boi\n\n" + logo
+                + "    Hit me up with those deets and let's vibe together!\n" + LINE);
+    }
+
+    /**
+     * Sends a message for terminating the program.
+     * Terminates the program by setting a flag hasEnded.
+     */
+    public void bye() {
```

Comment:
maybe these could be named `showHello()` and `showBye()`? as long as a verb is included 👍